includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -803,8 +801,6 @@ public function process_payment( $order_id ) {
 				$order->add_order_note( $note );
 			}
 
-			UPE_Payment_Gateway::remove_upe_payment_intent_from_session();
```

Comment:
Relocating this into UPE-gateway - intent caching is UPE specific and should not be added here in the first place.